src/tools/plan-generator.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import {Recipe} from '../runtime/recipe/recipe.js';
+import {Type} from '../runtime/type.js';
+import {Particle} from '../runtime/recipe/particle.js';
+import {KotlinGenerationUtils} from './kotlin-generation-utils.js';
+import {HandleConnection} from '../runtime/recipe/handle-connection.js';
+import {StorageKey} from '../runtime/storageNG/storage-key.js';
+
+const ktUtils = new KotlinGenerationUtils();
+
+export class PlanGeneratorError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = 'PlanGeneratorError';
+  }
+}
+
+/** Generates plan objects from resolved recipes. */
+export class PlanGenerator {
+  constructor(private resolutions: Recipe[], private scope: string = 'arcs.core.data') {
```

Comment:
resolutions is a bit unusual names. What about "plans" ?

Reply:
better still: `resolvedRecipes`.